### DISCUSSION TITLE
connect github repository via deploy keys

### DISCUSSION THREAD

**Discussion by stephanclaus** [apache/airflow#26780] [Q&A]
Hi everyone,

I was checking the [airflow documentation](https://airflow.apache.org/docs/apache-airflow-providers-github/stable/connections/github.html) on how to connect to our github repository but the proposed way would require us to create a fake github service account user if we want to prevent user dependencies for this connection.

I know that other tools like jenkins have support for [deployment keys](https://docs.github.com/en/developers/overview/managing-deploy-keys) but could not find anything like this for airflow.

Is this possible as well in Airflow 2.3.x?

Thank you for your help, any hint is appreciated.

**Comment by potiuk** [comment]
Have you tried to use the deployment key as access token? I believe GitHub does not distinguish those,

Otherwise if it does not work - the easiest way wil be to extend the classes and possibly contribute it back. Extending Operators/Hooks if they miss certain functionality is first-class-citizen in Airflow and our users are expected to do that (and if they are nice to the community they often contribute those changes back - we have more than 2200 contributors - often users like you who missed a functionality and added it first for themselves and contributed back.
